src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,214 @@
+import { Epsg } from '@basemaps/geo';
+import { fsa } from '@chunkd/fs';
+import { CogTiff } from '@cogeotiff/core';
+import { command, option, positional, string } from 'cmd-ts';
+import { StacCollection, StacItem } from 'stac-ts';
+
+import { CliInfo } from '../../cli.info.js';
+import { logger } from '../../log.js';
+import { config, createTiff, registerCli, verbose } from '../common.js';
+import { dataCategories, regions } from './path.constants.js';
+
+export const commandGeneratePath = command({
+  name: 'generate-path',
+  description: 'Generate target path from collection metadata',
+  version: CliInfo.version,
+  args: {
+    config,
+    verbose,
+
+    targetBucketName: option({
+      type: string,
+      long: 'target-bucket-name',
+      description: 'Target bucket name, e.g. nz-imagery',
+    }),
+
+    source: positional({
+      type: string,
+      displayName: 'path',
+      description: 'path to source data where collection.json file is located',
+    }),
+  },
+
+  async handler(args) {
+    registerCli(this, args);
+    const startTime = performance.now();
+
+    if (args.targetBucketName.startsWith('s3://')) {
+      throw new Error(
```

Comment:
We could just remove the `s3://` part from the front of the string. (then if its still not valid they would get the actual error.